gapic/src/main/com/google/gapid/perfetto/models/Selection.java

```diff
@@ -17,95 +17,149 @@
 
 import static com.google.gapid.util.MoreFutures.transform;
 import static com.google.gapid.util.MoreFutures.transformAsync;
-import static java.util.Arrays.stream;
-import static java.util.stream.Collectors.joining;
 
-import com.google.common.collect.Iterables;
 import com.google.common.collect.Maps;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
+import com.google.gapid.perfetto.models.SliceTrack.Slice;
+import com.google.gapid.perfetto.models.ThreadTrack.StateSlice;
 import com.google.gapid.perfetto.views.MultiSelectionView;
 import com.google.gapid.perfetto.views.State;
 
+import org.eclipse.swt.SWT;
 import org.eclipse.swt.widgets.Composite;
 
+import java.util.Iterator;
 import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
 
 /**
  * Data about the current selection in the UI.
  */
-public interface Selection {
+public interface Selection<Key> {
   public String getTitle();
+  public boolean contains(Key key);
   public Composite buildUi(Composite parent, State state);
-  public default void mark(@SuppressWarnings("unused") State state) { /* do nothing */ }
+  public default void markTime(@SuppressWarnings("unused") State state) { /* do nothing */ }
   public default void zoom(@SuppressWarnings("unused") State state) { /* do nothing */ }
 
-  public static class MultiSelection implements Selection {
-    private final Selection[] selections;
+  @SuppressWarnings("rawtypes")
+  public static final Selection EMPTY_SELECTION = new EmptySelection<>();
 
-    public MultiSelection(Selection[] selections) {
-      this.selections = selections;
-    }
+  @SuppressWarnings("unchecked")
+  public static <K> Selection<K> emptySelection() {
+      return EMPTY_SELECTION;
+  }
 
+  public static class EmptySelection<K> implements Selection<K> {
```

Comment:
private

Reply:
For usage it should be private, but unfortunately changing it to private here will trigger grammar error "The interface member type EmptySelection can only be public". :(